src/migmose/mig/reducednestednachrichtenstruktur.py

```diff
@@ -2,16 +2,135 @@
+                    )
+
+            # Compile the unique list from the seen dictionary after recursive processing to avoid circular reference
+            if depth == 0:  # Only compile on the initial call, not recursive ones
+                result = [seen[key] for key in seen]
+            return result
+
+        def build_segment_count_dict(segment_groups: list[NestedNachrichtenstruktur | None]):
+            segment_count_dict: dict = {}
+            for sg in segment_groups:
+                if sg is not None:
+                    name = get_identifier(sg.header_linie)
+                    count = count_segments(sg)
+
+                    # Check if the current segment group's count is greater than the stored count
+                    if name in segment_count_dict:
+                        existing_count, _ = segment_count_dict[name]
+                        if count > existing_count:
+                            segment_count_dict[name] = (count, sg)
+                    else:
+                        segment_count_dict[name] = (count, sg)
+
+                    # Process nested segment groups recursively and update the dictionary
+                    nested_counts = build_segment_count_dict(sg.segmentgruppen)
+                    for nested_name, (nested_count, nested_sg) in nested_counts.items():
+                        if nested_name in segment_count_dict:
+                            existing_count, _ = segment_count_dict[nested_name]
+                            if nested_count > existing_count:
+                                segment_count_dict[nested_name] = (nested_count, nested_sg)
+                        else:
+                            segment_count_dict[nested_name] = (nested_count, nested_sg)
+
+            return segment_count_dict
+
+        data = ReducedNestedNachrichtenstruktur()
+        # Start processing the top-level segments
+        if nachrichten_struktur.segmente is not None:
+            data.segmente = process_segments(nachrichten_struktur.segmente)
+
+        # Process segment groups recursively
+        if nachrichten_struktur.segmentgruppen is not None:
+            segment_count_dict = build_segment_count_dict(nachrichten_struktur.segmentgruppen)
+            data.segmentgruppen = process_segmentgruppen(nachrichten_struktur.segmentgruppen, segment_count_dict)
+
+        return data
+
+    def to_json(self, message_type: EdifactFormat, output_dir: Path) -> dict[str, Any]:
+        """
+        writes the reduced NestedNachrichtenstruktur as json
+        """
+        output_dir.mkdir(parents=True, exist_ok=True)
+        file_path = output_dir.joinpath(f"{message_type}_reduced_nested_nachrichtenstruktur.json")
+        structured_json = self.model_dump()
+        with open(file_path, "w", encoding="utf-8") as json_file:
+            json.dump(structured_json, json_file, indent=4)
+        logger.info(f"Wrote reduced nested Nachrichtenstruktur for {message_type} to {file_path}")
```

Comment:
Please avoid the f-string syntax in log message. 
Use the `%s` placeholder instead.
Why? See here: https://chat.openai.com/share/c3c36228-ba52-40cc-a223-95556c5447b4

Reply:
warum schlägt pylint da nicht an?

Reply:
https://github.com/Hochfrequenz/migmose/pull/41/commits/38c7c2422e0ede415275ac23898ca4dabf6973fb

Reply:
Lintter will nicht:
![image](https://github.com/Hochfrequenz/migmose/assets/66730658/68b7196d-c85f-4661-bb80-ff440ffe7c4d)